Comparison Engine/Comparison Engine/Child Forms/UserContribution.cs

```diff
@@ -5,32 +5,126 @@
 using System.Data;
 using System.Drawing;
 using System.Text;
+using System.Text.RegularExpressions;
 using System.Windows.Forms;
 
 namespace Comparison_Engine.Child_Forms
 
 {
     public partial class UserContribution : Form
     {
-        public UserContribution(Drink drink, Bar bar)
+        private DrinkManager drinkmanager = DrinkManager.Instance;
+        private BarManager barmanager = BarManager.Instance;
+        public UserContribution()
         {
+            Dictionary<int, float> barChoices = new Dictionary<int, float>();
             InitializeComponent();
+            PopulateComboBoxBar();
+            PopulateComboBoxDrink();
+            comboBoxBar.SelectedIndexChanged += new EventHandler(ComboBoxBarSelectionChange);
+        }
+
+        private void ComboBoxBarSelectionChange(object sender, EventArgs e)
+        {
+            var selectedBar = (Bar)comboBoxBar.SelectedItem;
+            List<Drink> drinklist = new List<Drink>();
+            foreach (var drink in selectedBar.availableDrinks)
+            {
+                drinklist.Add(drinkmanager.GetDrinkByID(drink.Key));
+            }
+            comboBoxDrink.DataSource = new BindingSource(drinklist, null);
         }
 
-        
         private void buttonAddDrink_Click(object sender, System.EventArgs e)
         {
+            if (textBoxDrink.Text == "" || comboBoxBar2.SelectedItem == null )
+            {
+                return;
```

Comment:
Would be nice if the user got some kind of message/visual feedback if they did not correctly add a drink